### ISSUE TITLE
Missing final boundary for empty multipart entities

### ISSUE THREAD

**Issue by arnohaase** [akka/akka-http#1257]
When rendering a multipart entity with no parts, BodyPartRenderer does not write the final boundary. This results in an entity that Akka HTTP can not parse:

```
val entity = Multipart.General
  .Strict(MediaTypes.`multipart/mixed`, Vector.empty)
  .toEntity("asdf")

val um = MultipartUnmarshallers.defaultMultipartGeneralUnmarshaller
val unmarshalled = Await.result(um(entity), 10.seconds) // triggers an exception

println (unmarshalled)
```

The suggested fix is to move renderFinalBoundary out of the if block in class BodyPartRenderer.

```
  def strict(parts: immutable.Seq[Multipart.BodyPart.Strict], boundary: String,
             partHeadersSizeHint: Int, log: LoggingAdapter): ByteString = {
    val r = new ByteStringRendering(partHeadersSizeHint)
    if (parts.nonEmpty) {
      for (part ← parts) {
        renderBoundary(r, boundary, suppressInitialCrLf = part eq parts.head)
        renderEntityContentType(r, part.entity)
        renderHeaders(r, part.headers, log)
        r ~~ part.entity.data
      }
      // <--- renderFinalBoundary removed here
    }
    renderFinalBoundary(r, boundary)   // <--- ... and moved here
    r.get
  }
```

**Comment by arnohaase** [comment]
This being my first bug report - I am unsure as to the best procedure. Should I create a pull request?

**Comment by raboof** [comment]
Hi Arno, Welcome!

Yes, a pull request would be great! You can read more about our contribution process in CONTRIBUTING.md - it's quite a bit of text, but don't worry, most of it is pretty light once you get the hang of it. It'd be great if you could reproduce the problem in a unit test. If you get stuck anywhere feel free to drop by on gitter at https://gitter.im/akka/dev - we're always happy to help.

**Comment by arnohaase** [comment]
Thanks - I'll do that over the weekend.

**Comment by jrudolph** [comment]
Fixed by #1260.
